app/controllers/api/api_controller.rb

```diff
@@ -0,0 +1,28 @@
+module Api
+  class ApiController < ApplicationController
+    skip_before_action :verify_authenticity_token
+
+    def posts_index
+      @user = User.find_by_api_token(request.headers['X-AUTH-TOKEN'])
+      @posts = Post.where(user_id: @user.id)
+      render json: @posts
+    end
+
+    def list_comments
+      @user = User.find_by_api_token(request.headers['X-AUTH-TOKEN'])
+      @comments = Comment.where(user_id: @user.id)
+      render json: @comments
+    end
+
```

Comment:
- [ ] Kindly note that this should return comments for the given post not all comments for the current user. Hint :point_down: 

### Example
```rb
@comments = Comment.where(user_id: @user.id, post_id: params[:post_id])
```

Reply:
Thank you for the suggestion 👍 .  I have implemented the changes